app/client/pages/user/[user_id].js

```diff
@@ -242,10 +306,10 @@ export default function profile(props) {
                         <TextField
```

Comment:
Knowledge and interests should be tag based components. You can use the code here: https://stackoverflow.com/questions/71585321/create-tag-using-autocomplete-material-ui-when-user-input-anything.   When viewing public profile disable the input